python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/testing_framework.py

```diff
@@ -0,0 +1,303 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+from typing import Any, List, Iterable, Optional, Callable
+
+import torch
+from torch import Tensor
+
+# ==============================================================================
+# Shape, dtype, and decomposition function testing infrastructure.
+# ==============================================================================
+
+# We expect all functions to be adequately tested. For functions
+# implemented with upstream helpers, additional testing is usually not needed.
+# But for functions that are authored/maintained by the Torch-MLIR
+# project, we expect adequate testing.
+#
+# To do this, we provide decorators `@check_shape/dtype/decomposition_function`
```

Comment:
nit: same comment here about using the full identifiers to aid greppability/sedability